### PR TITLE
Find visible elements

### PR DESCRIPTION
**Pull request by remcowesterhoud**
- [ X] By placing an `X` in the preceding checkbox, I verify that I have signed the [Contributor License Agreement](https://github.com/SeleniumHQ/selenium/blob/master/CONTRIBUTING.md#step-6-sign-the-cla)

Adds an expected conditions for finding visible elements as list. Right now this is only possible for the presence of elements.

### CODE REVIEW

**Comment by lukeis** [comment]
Can you also add a test for this?

https://github.com/seleniumhq/selenium/blob/master/py/test/selenium/webdriver/common/webdriverwait_tests.py

The pages that these tests are loading are here:
https://github.com/seleniumhq/selenium/blob/master/common/src/web/

File: py/selenium/webdriver/support/expected_conditions.py [hunk 1]
```diff
@@ -101,6 +101,22 @@ def __init__(self, locator):
...
+    locator is used to find the element
+    returns the list of WebElements once they are located
+    """
+    def __init__(self, locator):
+        self.locator = locator
+
+    def __call__(self, driver):
+        visible_elements = []
+        for element in _find_elements(driver, self.locator):
+            if _element_if_visible(element) is not False:
```
Comment (cgoldberg) [inline py/selenium/webdriver/support/expected_conditions.py]: I think this line would be better written as:
`if not _element_if_visible(element):`

File: py/selenium/webdriver/support/expected_conditions.py [hunk 2]
```diff
@@ -101,6 +101,22 @@ def __init__(self, locator):
...
+class visibility_of_all_elements_located(object):
+    """ An expectation for checking that there is at least one element visible
+    on a web page.
+    locator is used to find the element
+    returns the list of WebElements once they are located
+    """
+    def __init__(self, locator):
+        self.locator = locator
+
+    def __call__(self, driver):
```
Comment (cgoldberg) [inline py/selenium/webdriver/support/expected_conditions.py]: can't you use a list comprehension to replace your entire `__call__` method?

```
def __call__(self, driver):
	return [element for element in _find_elements(driver, self.locator) if _element_if_visible(element)]
```

**Comment by remcowesterhoud** [comment]
I've added the tests and replaced the call method with the list comprehension @cgoldberg mentioned.

File: py/test/selenium/webdriver/common/webdriverwait_tests.py [hunk 3]
```diff
@@ -78,6 +78,27 @@ def testShouldFailToFindElementsWhenExplicitWaiting(self):
...
+        add_hidden.click()
+
+        elements = WebDriverWait(self.driver, 2).until(EC.visibility_of_all_elements_located((By.CLASS_NAME, "redbox")))
+        self.assertTrue(len(elements) == 2)
+
+    def testShouldFailToFindVisibleElementsWhenExplicitWaiting(self):
+        self._loadPage("hidden_partially")
+        try:
+            elements = WebDriverWait(self.driver, 0.7).until(EC.visibility_of_all_elements_located((By.CLASS_NAME, "redbox")))
+        except TimeoutException as e:
```
Comment (cgoldberg) [inline py/test/selenium/webdriver/common/webdriverwait_tests.py]: you could assert the exception is raised:
https://docs.python.org/2.7/library/unittest.html#unittest.TestCase.assertRaises

**Comment by remcowesterhoud** [comment]
@cgoldberg I've changed it. I did it this way because I saw it being done in other testcases the same way.

**Comment by lukeis** [comment]
awesome, thanks for the patch! (and the reworks :) )

**Pull request merged**